packages/vanilla/src/complex/TableArrayControl.tsx

```diff
@@ -110,42 +114,71 @@ class TableArrayControl extends RendererComponent<TableProps, void> {
+            <th>
+              Valid
+            </th>
           </tr>
           </thead>
           <tbody>
           {
             (!data || !Array.isArray(data) || data.length === 0) ?
-              <tr><td>No data</td></tr> : data.map((_child, index) => {
-              const childPath = Paths.compose(path, `${index}`);
+              <tr><td>No data</td></tr> :
+              data.map((_child, index) => {
+                const childPath = Paths.compose(path, `${index}`);
+                const errorsPerEntry = _.filter(
+                  childErrors,
+                  error => error.dataPath.startsWith(childPath)
+                );
 
-              return (
-                <tr key={childPath}>
-                  {
-                    _.chain(scopedSchema.properties)
-                      .keys()
-                      .filter(prop => scopedSchema.properties[prop].type !== 'array')
-                      .map((prop, idx) => {
-                        return (
-                          <td key={Paths.compose(childPath, idx.toString())}>
-                            <DispatchField
-                              schema={scopedSchema}
-                              uischema={createControlElement(prop)}
-                              path={childPath}
-                            />
-                          </td>
-                        );
-                      })
-                      .value()
-                  }
-                </tr>
-              );
-            })
+                return (
+                  <tr key={childPath}>
+                    {
+                      scopedSchema.properties ?
+                        _.chain(scopedSchema.properties)
+                          .keys()
+                          .filter(prop => scopedSchema.properties[prop].type !== 'array')
+                          .map(prop => {
+                            const childPropPath = Paths.compose(childPath, prop.toString());
+
+                            return (
+                              <td key={childPropPath}>
+                                <DispatchField
+                                  schema={scopedSchema}
+                                  uischema={createControlElement(prop)}
+                                  path={childPath}
+                                />
+                              </td>
+                            );
+                          })
+                          .value() :
+                        <td key={Paths.compose(childPath, index.toString())}>
+                          <DispatchField
+                            schema={scopedSchema}
+                            uischema={createControlElement()}
+                            path={childPath}
+                          />
+                        </td>
+                    }
+                    <td>
+                      {
+                        errorsPerEntry ?
+                          <span style={{ color: 'red'}}>
```

Comment:
shouldn't this be a class only, with hardcoding this to red, one can never change this otherwise?